CMakeLists.txt

```diff
@@ -0,0 +1,342 @@
+cmake_minimum_required(VERSION 3.28)
+
+cmake_policy(SET CMP0048 NEW)
+cmake_policy(SET CMP0091 NEW)
```

Comment:
These two lines doesn't appear to be needed if cmake_minimum_required is 3.28?

Reply:
Hi.

In cmake, most [policies](https://cmake.org/cmake/help/book/mastering-cmake/chapter/Policies.html) default to the `OLD` behaviour, in order to remain compatible to previous behaviour when people bump up `cmake_mininum_required_version`.

[CMP0048](https://cmake.org/cmake/help/latest/policy/CMP0048.html)'s last paragraph states that unless explicitly set, the old behaviour will be used.
The same goes for [CMP0091](https://cmake.org/cmake/help/latest/policy/CMP0091.html).

Reply:
No?

[CMP0091](https://cmake.org/cmake/help/v3.28/policy/CMP0091.html) page 

> Use the [cmake_policy()](https://cmake.org/cmake/help/v3.28/command/cmake_policy.html#command:cmake_policy) command to set it to OLD or NEW explicitly.

The [cmake_minimum_required](https://cmake.org/cmake/help/v3.28/command/cmake_minimum_required.html#policy-settings) page from v3.28 says

> All policies known to the running version of CMake and introduced in the <min> (or <max>, if specified) version or earlier will be set to use NEW behavior.

I think this means `cmake_minimum_required(VERSION 3.28)` will set `NEW` for all the policies introduced in `VERSION <= 3.28`.